internal/log_analysis/datacatalog_updater/datacatalog/create_tables.go

```
@@ -97,17 +82,15 @@ func (h *LambdaHandler) fetchAllDeployedLogTypes(ctx context.Context) ([]string,
 }
 
 // Resolves the tables for the provided log types.
-// Note that this will return only the BASE tables (tables in for panther_logs, panther_cloudsecurity datbases) but not any
+// Note that this will return only the BASE tables (tables in for panther_logs and panther_cloudsecurity databases) but not any
 // downstream tables e.g. panther_rule_matches, panther_rule_errors
-func resolveTables(ctx context.Context, r logtypes.Resolver, names ...string) ([]*awsglue.GlueTableMetadata, error) {
+func resolveTables(ctx context.Context, r logtypes.Resolver, names ...string) []*awsglue.GlueTableMetadata {
 	var out []*awsglue.GlueTableMetadata
 	for _, name := range names {
 		entry, err := r.Resolve(ctx, name)
-		if err != nil {
-			return nil, err
-		}
-		if entry == nil {
-			return nil, errors.Errorf("unresolved log type %q", name)
+		if err != nil || entry == nil { // don't fail whole operation if inconsistent data...
+			zap.L().Error("unresolved log type", zap.String("name", name), zap.Error(err))
```

Logging an error here is *not* the correct way to handle an unresolved log type here.
The function should return the error to the caller and it should be handled there.
This introduces a global dependency and silently skips the error.
`zap.L()` usage is almost always the wrong way to handle an error.
Logging of errors should only happen 'close' to the top level of handler code.

---

So, the issue is I was able to create some inconsistent data (messing in my account) and this caused everything to fail because this function failed ... debugging this was complicated by the fact that the lambda lost all logging of errors , so there were no alarms or logs.

@alxarch If we do not log here AND we do not want to fail, how do you suggest passing this info back to the caller? We do want to know about this situation but we do not want the whole system to fail. I would say an error is not the right way, since the function did not fail.

---

@alxarch I made the function return a slice of warning errors and all callers up to the handler, where it is logged. Seems like over engineering but functionally the same so we have these properties:

 - one bad logType does not cause the whole system to collapse
 - we get paged when there is a bad logType

---

I think having an unresolved log type when creating views/tables *is* an irrecoverable error. The system would not collapse. The views just won't get updated.

---

If we organize the management of log types a bit better by not relying on Glue catalog state for our system state we could avoid even having to resolve the tables multiple times. If you consider when the `resolveTables` function is used we should not have any unresolved log types at that point. It would mean something is really wrong. I would still suggest returning an error (even if using `multierror.Append` to collect all such errors) and fail the operation immediately.

---

The `multierr.Append` worked but was messy. I found the clean solution, I move the logging into the resolver that is created in the `main` of the lambda.